test/services/datasources/hooks.test.js

```diff
@@ -0,0 +1,114 @@
+const { expect } = require('chai');
+
+const app = require('../../../src/app');
+const testObjects = require('../helpers/testObjects')(app);
+const { validateData, updatedBy, createdBy } = require('../../../src/services/datasources/hooks');
+
+describe('datasources hooks', () => {
+	describe('validateData', () => {
+		let server;
+		before((done) => {
+			server = app.listen(0, done);
+		});
+
+		after((done) => {
+			server.close(done);
+		});
+
+		it('filters and updates data correctly', async () => {
+			const fut = validateData;
```

Comment:
I guess `validateData` is short enough to write once per test. You could extract this to the describe block though